Sources/Document/CRDT/CRDTTree.swift

```diff
@@ -67,6 +67,65 @@ struct CRDTTreePos: Equatable {
 }
 
 extension CRDTTreePos {
+    /**
+     * `fromTreePos` creates a new instance of CRDTTreePos from the given TreePos.
+     */
+    static func fromTreePos(pos: TreePos<CRDTTreeNode>) -> CRDTTreePos {
+        let offset = Int(pos.offset)
+        var node = pos.node
+        var leftNode: CRDTTreeNode!
+
+        if node.isText {
+            if node.parent?.children[0] === node, offset == 0 {
+                leftNode = node.parent
+            } else {
+                leftNode = node
+            }
+
+            node = node.parent!
+        } else {
+            if offset == 0 {
+                leftNode = node
+            } else {
+                leftNode = node.children[offset - 1]
+            }
+        }
+
+        return CRDTTreePos(parentID: node.id, leftSiblingID: CRDTTreeNodeID(createdAt: leftNode.createdAt, offset: leftNode.offset + Int32(offset)))
+    }
+
+    /**
+     * `toTreeNodes` converts the pos to parent and left sibling nodes.
+     * If the position points to the middle of a node, then the left sibling node
+     * is the node that contains the position. Otherwise, the left sibling node is
+     * the node that is located at the left of the position.
+     */
+    func toTreeNodes(tree: CRDTTree) throws -> (CRDTTreeNode, CRDTTreeNode) {
+        let parentID = self.parentID
+        let leftSiblingID = self.leftSiblingID
+        let parentNode = tree.findFloorNode(parentID)
+        let leftNode = tree.findFloorNode(leftSiblingID)
+        guard let parentNode, var leftNode else {
+            throw YorkieError.unexpected(message: "cannot find node at \(self)")
+        }
+
+        /**
+         * NOTE(hackerwins): If the left node and the parent node are the same,
+         * it means that the position is the left-most of the parent node.
+         * We need to skip finding the left of the position.
+         */
+        if leftSiblingID != parentID,
```

Comment:
The equivalent JS code doesn't compare between leftSiblingID and parentID.
Do ignore if this was intentional.

Reply:
These codes are added in https://github.com/yorkie-team/yorkie-js-sdk/pull/704
This code will eventually be added.